api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -105,6 +107,16 @@ public CohortDefinition getAnyEncounterDuringPeriod() {
         return new MappedParametersCohortDefinition(cd, "onOrAfter", "startDate", "onOrBefore", "endDate");
     }
 
+    @DocumentedDefinition("codedObsSearchAdvanced")
+    public CohortDefinition getPatientsHavingObs() {
+        CodedObsCohortDefinition cd = new CodedObsCohortDefinition();
```

Comment:
I understand now @mseaton, I'll keep working on this branch to implement that.
Meanwhile, I've updated the method name to match the library key like you recommended.